### PR TITLE
Update controller logging, fix session leak:

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

This adds logging to show metadata about all bmc calls. Refactored logging in all controllers to pull the logger from ctx. This is because the controller runtime enriches the logs with a lot of detail that we can take advantage of.

Removed the zap logger in favor of zerolog because its of its smaller dependency footprint and so that logs go to stdout instead of stderr by default.

Fixed a session leak in the task controller. This was happening because the defer to close the session wasn't always being called.

## Why is this needed
When something doesn't work there's almost no logging to help with troubleshooting.

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: controllers/task_controller.go [hunk 1]
```diff
@@ -218,11 +235,13 @@ func (r *TaskReconciler) checkTaskStatus(ctx context.Context, task bmcv1alpha1.A
...
 		case bmcv1alpha1.PowerOn:
-			if bmcv1alpha1.On != state {
+			if state != bmcv1alpha1.On {
+				log.Info("requeuing task", "requeueAfter", powerActionRequeueAfter)
 				return ctrl.Result{RequeueAfter: powerActionRequeueAfter}, nil
 			}
 		case bmcv1alpha1.PowerHardOff, bmcv1alpha1.PowerSoftOff:
-			if bmcv1alpha1.Off != state {
+			if state != bmcv1alpha1.Off {
+				log.Info("requeuing task", "requeueAfter", powerActionRequeueAfter)
```
Comment (chrisdoherty4) [inline controllers/task_controller.go:244]: Iirc, the controller will log when the reconciler returns a re-queue result.

File: main.go [hunk 2]
```diff
@@ -51,25 +59,52 @@ func init() {
...
 			"Enabling this will ensure there is only one active controller manager.")
-	flag.StringVar(&kubeAPIServer, "kubernetes", "", "The Kubernetes API URL, used for in-cluster client construction.")
-	flag.StringVar(&kubeconfig, "kubeconfig", "", "Absolute path to the kubeconfig file.")
-	flag.StringVar(&kubeNamespace, "kube-namespace", "", "Namespace that the controller watches to reconcile objects.")
+	fs.StringVar(&kubeAPIServer, "kubernetes", "", "The Kubernetes API URL, used for in-cluster client construction.")
+	fs.StringVar(&kubeconfig, "kubeconfig", "", "Absolute path to the kubeconfig file.")
+	fs.StringVar(&kubeNamespace, "kube-namespace", "", "Namespace that the controller watches to reconcile objects.")
+	cli := &ffcli.Command{
+		Name:       appName,
+		ShortUsage: "Run Boots server for provisioning",
```
Comment (chrisdoherty4) [inline main.go:100]: Did you mean to introduce ff?

**Review by chrisdoherty4 — COMMENTED** [review]
Thanks for the changes.

File: main.go (hunk 2 above)
Comment (jacobweinstock) [reply main.go:100]: whoops! no. thanks for the catch. will remove.

File: controllers/task_controller.go (hunk 1 above)
Comment (jacobweinstock) [reply controllers/task_controller.go:244]: sure, i will remove.

**Review by chrisdoherty4 — APPROVED** [review]

**Comment by jacobweinstock** [comment]
@mergifyio queue

**Pull request merged**